### PR TITLE
RDK-38400: Support for Checkpoint and Restore in RDKShell

### PR DESCRIPTION
**Pull request by adrianM27**

### CODE REVIEW

File: RDKShell/RDKShell.cpp [hunk 1]
```diff
@@ -141,6 +141,8 @@ const string WPEFramework::Plugin::RDKShell::RDKSHELL_METHOD_GET_AV_BLOCKED_APPS
 const string WPEFramework::Plugin::RDKShell::RDKSHELL_METHOD_KEY_REPEAT_CONFIG = "keyRepeatConfig";
 const string WPEFramework::Plugin::RDKShell::RDKSHELL_METHOD_GET_GRAPHICS_FRAME_RATE = "getGraphicsFrameRate";
 const string WPEFramework::Plugin::RDKShell::RDKSHELL_METHOD_SET_GRAPHICS_FRAME_RATE = "setGraphicsFrameRate";
+const string WPEFramework::Plugin::RDKShell::RDKSHELL_METHOD_CHECKPOINT = "checkpoint";
```
Comment (anand-ky) [inline RDKShell/RDKShell.cpp:144]: Pls update JSON API spec RDKShell.json with the new APIs and events so that it gets updated in documentation.

**Review by anand-ky — CHANGES_REQUESTED** [review]

File: RDKShell/RDKShell.cpp [hunk 2]
```diff
@@ -5740,6 +5759,101 @@ namespace WPEFramework {
...
+                    std::cout << "ignoring checkpoint for " << callsign << " as it is being destroyed " << std::endl;
+                    status = false;
+                    response["message"] = "failed to checkpoint application, is being destroyed";
+                    returnResponse(status);
+                }
+
+                std::thread requestsThread =
+                std::thread([=]()
+                {
+                    auto systemServiceConnection = RDKShell::getThunderControllerClient();
```
Comment (binuinbaraj) [inline RDKShell/RDKShell.cpp:5793]: systemServiceConnection is misleading,  please rename this

File: RDKShell/RDKShell.cpp [hunk 3]
```diff
@@ -5740,6 +5759,101 @@ namespace WPEFramework {
...
+        {
+            LOGINFOMETHOD();
+            bool status = false;
+            if (parameters.HasLabel("callsign"))
+            {
+                std::string callsign = parameters["callsign"].String();
+                std::thread requestsThread =
+                std::thread([=]()
+                {
+                    auto systemServiceConnection = RDKShell::getThunderControllerClient();
```
Comment (binuinbaraj) [inline RDKShell/RDKShell.cpp:5834]: systemServiceConnection is misleading,  please rename this

File: RDKShell/RDKShell.cpp (hunk 2 above)
Comment (adrianM27) [reply RDKShell/RDKShell.cpp]: changed to thunderController

File: RDKShell/RDKShell.cpp (hunk 1 above)
Comment (adrianM27) [reply RDKShell/RDKShell.cpp:144]: Done

**Pull request merged**